javascript/modules/client/src/client-erc20.ts

```diff
@@ -1,703 +1,52 @@
+  IClientErc20Methods,
   IErc20PluginInstall,
-  IExecuteProposalParams,
-  IMintTokenParams,
-  IPluginSettings,
-  IProposalQueryParams,
-  IVoteProposalParams,
-  ProposalCreationSteps,
-  ProposalCreationStepValue,
-  ProposalMetadata,
-  ProposalSortBy,
-  SubgraphErc20Proposal,
-  SubgraphErc20ProposalListItem,
-  VoteProposalStep,
-  VoteProposalStepValue,
-  VoteValues,
 } from "./internal/interfaces/plugins";
-import { IDAO } from "@aragon/core-contracts-ethers";
-import { ClientCore } from "./internal/core";
-import {
-  DaoAction,
-  GasFeeEstimation,
-  IInterfaceParams,
-  IPluginInstallItem,
-  SortDirection,
-} from "./internal/interfaces/common";
+import { IClientErc20MethodsModule } from "./modules-erc20/methods";
 import { ContextPlugin } from "./context-plugin";
-import {
-  computeProposalStatusFilter,
-  isProposalId,
-  toErc20Proposal,
-  toErc20ProposalListItem,
-} from "./internal/utils/plugins";
-import {
-  decodeMintTokenAction,
-  decodeUpdatePluginSettingsAction,
-  encodeErc20ActionInit,
-  encodeMintTokenAction,
-  encodeUpdatePluginSettingsAction,
-  getFunctionFragment,
-} from "./internal/encoding/plugins";
-import {
-  bytesToHex,
-  GraphQLError,
-  InvalidAddressError,
-  InvalidAddressOrEnsError,
-  InvalidProposalIdError,
-  NoProviderError,
-  Random,
-} from "@aragon/sdk-common";
-import { delay } from "./internal/temp-mock";
-import { isAddress } from "@ethersproject/address";
-import {
-  QueryErc20Proposal,
-  QueryErc20Proposals,
-} from "./internal/graphql-queries/proposal";
-import { formatEther } from "@ethersproject/units";
-import { QueryErc20PluginSettings } from "./internal/graphql-queries/settings";
-import { QueryToken } from "./internal/graphql-queries/token";
+import { IClientErc20EncodingModule } from "./modules-erc20/encoding";
+import { IClientErc20DecodingModule } from "./modules-erc20/decoding";
+import { IClientErc20EstimationModule } from "./modules-erc20/estimation";
+import { IPluginInstallItem } from "./internal/interfaces/common";
+import { encodeErc20ActionInit } from "./internal/encoding/plugins";
+import { ClientCore } from "./internal/core";
 
 // NOTE: This address needs to be set when the plugin has been published and the ID is known
 const PLUGIN_ID = "0x1234567890123456789012345678901234567890";
 /**
  * Provider a generic client with high level methods to manage and interact an ERC20 Voting plugin installed in a DAO
  */
 export class ClientErc20 extends ClientCore implements IClientErc20 {
+  public methods: IClientErc20Methods;
+  public encoding: IClientErc20Encoding;
+  public decoding: IClientErc20Decoding;
+  public estimation: IClientErc20Estimation;
+
   constructor(context: ContextPlugin) {
     super(context);
+    this.methods = new IClientErc20MethodsModule(context);
+    this.encoding = new IClientErc20EncodingModule(context);
+    this.decoding = new IClientErc20DecodingModule(context);
+    this.estimation = new IClientErc20EstimationModule(context);
```

Comment:
Could this be readonly? I think that if it is assiged on the constructor, we could have as an immutable property. Otherwise, having a getter might help too.

Reply:
Yes I we can have them as 
```ts
readonly methods
readonly encoding
readonly decoding
readonly estimation
```
and won't affect the usage

Reply:
Whatever that gives us 100% certainty that no external user will be able to reimplement stuff and front run the SDK. 

Is `readonly` a TS compile-time only thing? Or is it native pure JS that actually prevents writing to? If not, I would define a getter reading private properties behind a `Symbol`. But maybe this is scope for another PR